test/genSaleActions.test.ts

```diff
@@ -0,0 +1,407 @@
+                merkleTreeFileUri: "https://example.com/mintlist",
+                web3Provider: new ethers.providers.JsonRpcProvider(),
+                advanced: {
+                    merkleTreeFileIPFSHash: "QmHash",
+                    ipfsGateway: "https://ipfs.io/ipfs/",
+                },
+            };
+
+            nock("https://example.com").get("/mintlist").reply(404, mockMintlist);
+
+            nock("https://ipfs.io").get("/ipfs/QmHash").reply(200, mockMintlist);
+
+            const result = await getMintlist(config);
+            expect(result).to.deep.equal(mockMintlist);
+        });
+    });
+
+    describe("getSaleStatus", () => {
+        const mockSaleAsGenSale: GenSale = mocks.genSale as GenSale; // Cast to GenSale to access types
+
+        it("Should return not started status if sale has not started", async () => {
+            mocks.currentSaleStarted = false;
+            const saleStatus = await getSaleStatus(mockSaleAsGenSale);
+            expect(saleStatus).to.equal(GenSaleStatus.NotStarted);
+        });
+
+        it("Should return claim status if sale has started and price is not set", async () => {
+            mocks.currentSaleStarted = true;
+            mocks.currentSalePrice = BigNumber.from(0);
+
+            const saleStatus = await getSaleStatus(mockSaleAsGenSale);
+
+            expect(saleStatus).to.equal(GenSaleStatus.ClaimSale);
+        });
+
+        it("Should return the Private status if sale has started and price is set", async () => {
+            mocks.currentSaleStarted = true;
+            mocks.currentSalePrice = salePriceValue;
+
+            const saleStatus = await getSaleStatus(mockSaleAsGenSale);
+
+            expect(saleStatus).to.equal(GenSaleStatus.PrivateSale);
+        });
+
+        it("Should return the not started status if the sale has a price set and sale has not started", async () => {
+            mocks.currentSaleStarted = false;
+            mocks.currentSalePrice = salePriceValue;
+
+            const saleStatus = await getSaleStatus(mockSaleAsGenSale);
+
+            expect(saleStatus).to.equal(GenSaleStatus.NotStarted);
+        });
+
+        it("Should return the ended status if the sale has started but all domains have been sold", async () => {
+            mocks.currentSaleStarted = true;
+            mocks.currentSalePrice = salePriceValue;
+            mocks.currentDomainsSold = BigNumber.from(50);
+            const saleStatus = await getSaleStatus(mockSaleAsGenSale);
+
+            expect(saleStatus).to.equal(GenSaleStatus.Ended);
+        });
+    });
+
+    describe("getSaleData", async () => {
+        it("Should return the sale data correctly", async () => {
+            const amountSold = 5;
+            const amountForSale = 50;
+            const transactionLimit = 10;
+            const saleConfigurationRaw = await getDefaultMockGenSaleConfiguration();
+
+            mocks.currentSaleConfiguration = saleConfigurationRaw;
```

Comment:
would be nice to call this mock as what it specifically represents. e.g `GenSaleContractMock` or `genSaleSdkInstanceMock`. we can also separate the big mock into individual objects so that we don't load unnecessary data and can easily tell what's what when reading the code. we can easily combine them if we want the full package.
this is with all the other comments about code structuring. a suggestion to the approach

Reply:
I think the single mock that Ryan developed was pretty easy to use, I think using multiple smaller mocks might be less manageable and confusing

Reply:
depends on preference, I guess. it's small functionality here and it's all enclosed into a single contract, so it makes sense to not overcomplicate.